### PR TITLE
Add test to reproduce bug in mainnet

### PR DESCRIPTION
**Pull request by mfornet**
Bug is already fixed in #548
Fix this test and add it.

### CODE REVIEW

File: contracts/near/eth-client/src/tests.rs [hunk 1]
```diff
@@ -461,3 +461,599 @@ fn predumped_block_can_be_added() {
     }
     bar.finish();
 }
+
+const DAGS_MERKLE_ROOT: &'static [&str] = &[
```
Comment (abacabadabacaba) [inline contracts/near/eth-client/src/tests.rs:465]: This may be a good use for `hex_literal` crate.

File: contracts/near/eth-client/src/tests.rs [hunk 2]
```diff
@@ -461,3 +461,599 @@ fn predumped_block_can_be_added() {
...
+    "0x9823e2e8911c5a0fb2ff6feb52be2c0f",
+    "0x4fa03dbaabcf99e71903f3177444bfaa",
+    "0xa4c9667d00998d262c47f6d2c79f43c2",
+    "0xb3e65ee26384d25bc7750b3f37e72883",
+    "0x7dea46d1a183f3070760eee0bf5fce3c",
+    "0xecb2963c22a757569fe659635f4b0243",
+    "0x9ed968203144981e6e697db052910c27",
+    "0x4aa6ca6ebef942d8766065b2e590fd32",
+];
+const BLOCK_HEADER: &'static [u8] = &[
```
Comment (abacabadabacaba) [inline contracts/near/eth-client/src/tests.rs:979]: This too.

File: contracts/near/eth-client/src/tests.rs [hunk 3]
```diff
@@ -461,3 +461,599 @@ fn predumped_block_can_be_added() {
...
...3872571033b9061463a3f684deb16966089e80a19ad586f9cfae6d4f2d77bd74b980f53d5be718d7e753afba","0x6640cab3ca71b0247ec3906eb9ea574dfff53ac75a9821920956595c01ff018d9a706c2644561eb6f5a1145e6a95d25719da16efd0060b1722d1d2d5d1644a39"],"proof":["0xd0ec45344a7091b405f1d2c962a3f5f3","0x7601ef319f9054567f0e859629443a41","0x0ac36d658461a7cd07de8894aef3a76d","0xc434344d03c524e546fbc79e6ca2ff7e","0x18077ed4f9fc888dfcc2fca5d89cff2c","0x7ee6112533a50b3bee77e21f25f19a48","0x9453be8fabaa453f889380ef22551953","0x50a97983dabc2af2c97834307bc56928","0x7ffa4a53e14df7b7fb18adfe6e9f1481","0x5c9158107d37c248a221ea1eff8a1498","0x1c53d206c610aecc9be2b9f0bf885ef4","0x2899b5ed788d33e08b7d4ff4e25cfb3e","0xa3082f77cb0294af5f986a08e7788fa3","0x7af554433b5c569e313b6dac844ad88a","0x371079722c88b2b02518b2a3433aaa32","0xe49a9195c0dc650906536d868fd15ba9","0x69e6092f996258aaa973a13e6ca20af8","0x59d640825d9feb78e8f6286409ff4275","0xe4318aeb26c130ef204e892a35995dc5","0xdc6f2fe4cb45c7fae9f39f4cc4c2eece","0x29f2e5c1b9bb174b88b63b5d6b8d105b","0x914b93e809e85da6f6af72afe6233e9f","0xcfd16af046f96eb2269c21c16498a65e","0x10b71b7523c69d2b5e95bfb0d90a17ab","0x158f1e4ac3fbda78f466298defde22b2","0x30a53f479a2ca1539ca2a5a71afae445"]}]"#;
```
Comment (abacabadabacaba) [inline contracts/near/eth-client/src/tests.rs:1034]: This may use `json!` macro from `serde_json` crate.

**Review by abacabadabacaba — CHANGES_REQUESTED** [review]

File: contracts/near/eth-client/src/tests.rs (hunk 2 above)
Comment (mfornet) [reply contracts/near/eth-client/src/tests.rs]: moved data to separate json file

File: contracts/near/eth-client/src/tests.rs (hunk 3 above)
Comment (mfornet) [reply contracts/near/eth-client/src/tests.rs]: moved data to separate json file

File: contracts/near/eth-client/src/tests.rs (hunk 1 above)
Comment (mfornet) [reply contracts/near/eth-client/src/tests.rs]: moved data to separate json file

**Comment by mfornet** [comment]
TODO: Compute again dag_nodes with the updated tool to make this test pass.

**Pull request closed without merging**